### PR TITLE
[MM-34836] Cleanse URLs of extra slashes on parsing, and add the image proxy case

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
When working with URLs that have multiple slashes in the pathname, if we tried to match them using one of our `is___URL` functions they wouldn't match due to the extra slashes.

This PR adds some code to cleanse the extra slashes out of URLs (since apparently the URL library doesn't do that on parsing), and explicitly defines the image proxy case such that those links will open in a browser.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34836

```release-note
NONE
```

### CODE REVIEW

File: src/common/utils/url.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ function parseURL(inputURL) {
         return inputURL;
     }
     try {
-        return new URL(inputURL);
```
Comment (Willyfrog) [inline src/common/utils/url.js:43]: can we add a unit test for the regex? so we can see what's a valid one and what are we filtering? that way if we find some other weird urls that cause us trouble we can make sure we don't reopen old problems

File: src/main/views/webContentEvents.js [hunk 2]
```diff
@@ -136,6 +136,12 @@ const generateNewWindowListener = (getServersFunction, spellcheck) => {
             return;
         }
 
+        // Image proxy case
+        if (parsedURL.pathname.match(/^\/api\/v[3-4]\/image/)) {
+            shell.openExternal(url);
+            return;
+        }
+
```
Comment (Willyfrog) [inline src/main/views/webContentEvents.js:144]: maybe we should send to `openExternal` anything that comes from the API? is there anything in API that should be displayed on the desktop?

**Review by Willyfrog — APPROVED** [review]
looking good, i left a couple of suggestions

File: src/main/views/webContentEvents.js (hunk 2 above)
Comment (devinbinnie) [reply src/main/views/webContentEvents.js:144]: I think we're blocking anything else that comes from the API, or it opens in a popup window. Not sure what the popup case is. I'd prefer to just handle individual cases as they come up and block other stuff unless it makes sense to allow it through. Just to reduce other potential security problems.

**Review by Willyfrog — APPROVED** [review]

File: src/main/views/webContentEvents.js [hunk 3]
```diff
@@ -136,6 +136,12 @@ const generateNewWindowListener = (getServersFunction, spellcheck) => {
             return;
         }
 
+        // Image proxy case
+        if (parsedURL.pathname.match(/^\/api\/v[3-4]\/image/)) {
```
Comment (hmhealey) [inline src/main/views/webContentEvents.js:140]: Strictly speaking, this API never existed in API v3, but we could always leave it for consistency

**Review by hmhealey — APPROVED** [review]

**Pull request merged**